include/h2o.h

```diff
@@ -919,6 +890,18 @@ typedef struct st_h2o_conn_callbacks_t {
      * debug state callback (optional)
      */
     h2o_http2_debug_state_t *(*get_debug_state)(h2o_req_t *req, int hpack_enabled);
+    /**
+     * returns number of closed idle connections
+     */
+    int (*close_idle_connection)(h2o_conn_t *conn);
+    /**
+     * shutdown of connection is requested (optional)
+     */
+    void (*request_shutdown)(h2o_conn_t *conn);
```

Comment:
I do not have a strong opinion here, but we can make this a required callback and let http1.c define an empty function. The cost of calling an empty function is often a no-op on x86.